chat.go

```diff
@@ -85,6 +85,17 @@ const (
 	FinishReasonNull          FinishReason = "null"
 )
 
+func (r FinishReason) MarshalJSON() ([]byte, error) {
+	switch r {
+	case FinishReasonNull, "":
+		return []byte("null"), nil
+	case FinishReasonStop, FinishReasonLength, FinishReasonFunctionCall, FinishReasonContentFilter:
```

Comment:
@yuikns Is this conditional branching necessary? It seems to do the same thing as the default. How about just having the default? It looks like we might forget to add to this statement if a new FinishReason is added in the future.

Reply:
@vvatanabe it is not necessary.
I was supposed to do some validation here, but i think it is not a good idea.

Reply:
@vvatanabe unfortunately if I just having the default will failed in CI checking. 

ref https://github.com/sashabaranov/go-openai/actions/runs/5684433643/job/15407141816?pr=449

```
  Error: missing cases in switch of type openai.FinishReason: openai.FinishReasonStop, openai.FinishReasonLength, openai.FinishReasonFunctionCall, openai.FinishReasonContentFilter (exhaustive)
  
  Error: issues found
  Ran golangci-lint in 9366ms
```

Reply:
@yuikns You can fix like this.

```go
func (r FinishReason) MarshalJSON() ([]byte, error) {
	if r == FinishReasonNull || r == "" {
		return []byte("null"), nil
	}
	return []byte(`"` + string(r) + `"`), nil // best effort to not break future API changes
}
```

Reply:
@vvatanabe I got caught in path dependency before. thank you, fixed

Reply:
LGTM